lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestSecurityToken.cs

```diff
@@ -0,0 +1,232 @@
+            Assert.Equal(FinalState.SUCCEEDED, state);
+        }
+
+        /// <summary>
+        /// Use HelloDriver in the test
+        /// </summary>
+        /// <returns></returns>
+        private IConfiguration GetDriverConfig()
+        {
+            return DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<HelloDriver>.Class)
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<HelloDriver>.Class)
+                .Build();
+        }
+
+        /// <summary>
+        /// Pull job final status until the Job is done
+        /// </summary>
+        /// <param name="jobSubmitionResult"></param>
+        /// <returns></returns>
+        private FinalState PullFinalJobStatus(IJobSubmissionResult jobSubmitionResult)
+        {
+            int n = 0;
+            var state = jobSubmitionResult.FinalState;
+            while (state.Equals(FinalState.UNDEFINED) && n++ < 100)
+            {
+                Thread.Sleep(3000);
+                state = jobSubmitionResult.FinalState;
+            }
+            return state;
+        }
+
+        /// <summary>
+        /// Get runtime configuration.
+        /// Bind tokens to YarnClientCnfiguration.
+        /// </summary>
+        /// <returns></returns>
+        private static IConfiguration GetRuntimeConfigurationForMultipleTokens()
+        {
+            var yarnClientConfigModule = YARNClientConfiguration.ConfigurationModule;
+            foreach (var t in CreateTestTokens())
+            {
+                yarnClientConfigModule = yarnClientConfigModule.Set(YARNClientConfiguration.SecurityTokenStr, t);
+            }
+
+            return Configurations.Merge(yarnClientConfigModule.Build(), TcpPortConfig());
+        }
+
+        /// <summary>
+        /// Get runtime configuration.
+        /// Bind token to YarnClientCnfiguration.
+        /// </summary>
+        /// <returns></returns>
+        private static IConfiguration GetRuntimeConfigurationForSingleToken()
+        {
+            var yarnClientConfigModule = YARNClientConfiguration.ConfigurationModule
+                .Set(YARNClientConfiguration.SecurityTokenStr, CreateTestToken());
+
+            return Configurations.Merge(yarnClientConfigModule.Build(), TcpPortConfig());
+        }
+
+        /// <summary>
+        /// Get runtime configuration and token with old approach
+        /// </summary>
+        /// <returns></returns>
+        private static IConfiguration GetRuntimeConfigurationBackwardComp()
+        {
+            var c = YARNClientConfiguration.ConfigurationModule
+                .Set(YARNClientConfiguration.SecurityTokenKind, Identifier1)
+                .Set(YARNClientConfiguration.SecurityTokenService, Identifier1)
+                .Build();
+
+            File.WriteAllText("SecurityTokenId", TokenKey1);
+            File.WriteAllText("SecurityTokenPwd", Password1);
```

Comment:
Don't use string literals. Call the proper methods of `REEFFileNames` class instead. (and don't call `new` on this class - use the injection)